tests/v2/validation/rbac/rbac_etcd_backup_test.go

```diff
@@ -0,0 +1,132 @@
+	rb.clusterName = client.RancherConfig.ClusterName
+	require.NotEmptyf(rb.T(), rb.clusterName, "Cluster name to install should be set")
+	clusterID, err := clusters.GetClusterIDByName(rb.client, rb.clusterName)
+	require.NoError(rb.T(), err, "Error getting cluster ID")
+	rb.cluster, err = rb.client.Management.Cluster.ByID(clusterID)
+	require.NoError(rb.T(), err)
+}
+
+func (rb *ETCDRbacBackupTestSuite) ValidateEtcdSnapshotCluster(role string) {
+
+	/* Testcase1 Verify cluster members - Owner/member,  Project members - Owner/member are able to perform snapshot action on clusters
+	Cluster Owner/Restricted Admin - Allowed
+	Cluster Member/Project Owner/Project Member - Not Allowed */
+	log.Infof("Creating a snapshot of the cluster as %v", role)
+
+	err := rke2.CreateSnapshot(rb.standardUserClient, rb.clusterName, 1, rb.namespace)
+
+	switch role {
+	case roleOwner, restrictedAdmin:
+		require.NoError(rb.T(), err)
+		// verify snapshot is created and is Active
+
+	case roleMember, roleProjectOwner, roleProjectMember:
+		require.Error(rb.T(), err)
+		strings.Contains(err.Error(), "Resource type [provisioning.cattle.io.cluster] is not updatable")
+	}
+}
+
+func (rb *ETCDRbacBackupTestSuite) TestETCDRbac() {
+	tests := []struct {
+		name   string
+		role   string
+		member string
+	}{
+		{"Cluster Owner", roleOwner, standardUser},
+		{"Cluster Member", roleMember, standardUser},
+		{"Project Owner", roleProjectOwner, standardUser},
+		{"Project Member", roleProjectMember, standardUser},
+		{"Restricted Admin", restrictedAdmin, restrictedAdmin},
+	}
+	for _, tt := range tests {
+		rb.Run("Set up User with Cluster Role "+tt.name, func() {
+			newUser, err := createUser(rb.client, tt.member)
+			require.NoError(rb.T(), err)
+			rb.standardUser = newUser
+			rb.T().Logf("Created user: %v", rb.standardUser.Username)
+			rb.standardUserClient, err = rb.client.AsUser(newUser)
+			require.NoError(rb.T(), err)
+
+			subSession := rb.session.NewSession()
+			defer subSession.Cleanup()
+
+			createProjectAsAdmin, err := createProject(rb.client, rb.cluster.ID)
+			rb.adminProject = createProjectAsAdmin
+			require.NoError(rb.T(), err)
+
+		})
+
+		rb.Run("Adding user as "+tt.name+" to the downstream cluster.", func() {
+			//Adding created user to the downstream clusters with the specified roles.
+
+			if tt.member == standardUser {
+				if strings.Contains(tt.role, "project") {
+					err := users.AddProjectMember(rb.client, rb.adminProject, rb.standardUser, tt.role)
+					require.NoError(rb.T(), err)
+				} else {
+					err := users.AddClusterRoleToUser(rb.client, rb.cluster, rb.standardUser, tt.role)
+					require.NoError(rb.T(), err)
+				}
+			}
+
+			relogin, err := rb.standardUserClient.ReLogin()
+			require.NoError(rb.T(), err)
+			rb.standardUserClient = relogin
+
+		})
+
+		rb.T().Logf("Starting validations for %v", tt.role)
+
+		rb.Run("Testcase1 - Validating if user is able to take Etcd snapshot of a cluster as the role "+tt.name, func() {
```

Comment:
nit - can we shorten this name? i.e. `testcase1 - take etcd snapshot as `

Reply:
done